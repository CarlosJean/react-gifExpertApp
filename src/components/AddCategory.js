import { useState } from "react";
import { PropTypes } from "prop-types";

export const AddCategory = ({setCategories}) => {

    const [category, setCategory] = useState('');

    const handleSubmit = (e)=>{
        e.preventDefault();        
        if (category.trim().length > 2) {
            setCategories( categories =>[category, ...categories]);
            setCategory('');
        }
    };

    const handleOnChange = (e) => {
        setCategory(e.target.value);         
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{category}</p>
            <input 
                type="text"
                placeholder="Ingrese algo"
                onChange={handleOnChange}
                value={category}
            />
            <button type="reset">Limpiar</button>
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}