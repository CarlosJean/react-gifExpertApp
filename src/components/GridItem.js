import PropTypes from "prop-types";

export const GridItem = ({title, url}) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <img 
                alt={title}
                src={url}
            />
            <p>{title}</p>
        </div>
    );
}

GridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}