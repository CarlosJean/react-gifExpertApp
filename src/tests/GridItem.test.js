import {shallow} from 'enzyme';

import { GridItem } from "../components/GridItem";

describe('Pruebas a <GridItem />', () => {

    const attrs = {
        url: 'https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=c8516906kvu2qatjv5ni3dvg8x9tquagc2o32ei4rql1oj8j&rid=giphy.gif&ct=g',
        title: 'One Punch Man GIF'
    };

    const wrapper = shallow(<GridItem {...attrs}/>);

    test('Debe mostrar <GridItem />', () => {      
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar un pÃ¡rrafo en el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(attrs.title);
    });

    test('La imagen debe tener el atributo src igual al url y el alt igual al title', () => {
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(attrs.url);
        expect(img.prop('alt')).toBe(attrs.title);
    })

    test('Debe tener la clase animate__fadeIn', ()=>{
        const div = wrapper.find('div');

        const includesAnimateFadeIn = div.prop('className')
        .includes('animate__fadeIn');

        expect(includesAnimateFadeIn).toBe(true);
    })
})