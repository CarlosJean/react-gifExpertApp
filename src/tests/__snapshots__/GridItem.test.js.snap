// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <GridItem /> Debe mostrar <GridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="One Punch Man GIF"
    src="https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=c8516906kvu2qatjv5ni3dvg8x9tquagc2o32ei4rql1oj8j&rid=giphy.gif&ct=g"
  />
  <p>
    One Punch Man GIF
  </p>
</div>
`;
